src/v/cloud_storage/segment_meta_cstore.h

```diff
@@ -118,6 +119,45 @@ class segment_meta_column_frame {
         }
     }
 
+    struct frame_tx_t {
+        typename encoder_t::tx_state inner;
+        segment_meta_column_frame<value_t, delta_t>& self;
+
+        frame_tx_t(
+          typename encoder_t::tx_state&& s,
+          segment_meta_column_frame<value_t, delta_t>& self)
+          : inner(std::move(s))
+          , self(self) {}
+
+        void add(const std::array<value_t, buffer_depth>& row) {
+            inner.add(row);
+        }
+        void commit() noexcept { self._tail->tx_commit(std::move(inner)); }
```

Comment:
can we make this safer by asserting that the frame_tx has not yet been committed? it also sort of looks like you can r-value qualify this too, as an alternative to the assertion.

Reply:
done